src/main/java/seedu/address/logic/sort/enums/SortTypeEnum.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.logic.sort.enums;
+
+/**
+ * Enum to represent the field to sort by in the sort.
+ */
+public enum SortTypeEnum {
+    TASK_NAME("task name"),
+    PRIORITY("priority"),
+    DEADLINE("deadline");
+
+    public static final String MESSAGE_CONSTRAINTS = "Oh no! I cannot understand the sort order: ";
+
+    private final String representation;
+
+    SortTypeEnum(String representation) {
+        this.representation = representation;
+    }
+
+    /**
+     * Returns the relevant {@code SortTypeEnum} based on the String input.
+     *
+     * @param sortOrder String input to parse
+     * @return {@code SortTypeEnum} object
+     * @throws IllegalArgumentException if the input String is not a valid input String value
+     */
+    public static SortTypeEnum of(String sortOrder) {
```

Comment:
Should the String input be sortType instead of sortOrder?

Reply:
Oh yikes, will fix in a future commit!